configuration/postgres/postgres.go

```diff
@@ -0,0 +1,296 @@
+/*
+Copyright 2021 The Dapr Authors
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package postgres
+
+import (
+	"context"
+	"database/sql"
+	"encoding/json"
+	"fmt"
+	"os"
+	"reflect"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/dapr/components-contrib/configuration"
+	"github.com/dapr/kit/logger"
+	"github.com/google/uuid"
+	"github.com/jackc/pgconn"
+	"github.com/jackc/pgx/v4/pgxpool"
+	_ "github.com/jackc/pgx/v4/stdlib"
+)
+
+type ConfigurationStore struct {
+	metadata             metadata
+	client               *pgxpool.Pool
+	logger               logger.Logger
+	subscribeStopChanMap sync.Map
+}
+
+const (
+	configtablekey               = "table"
+	connMaxIdleTimeKey           = "connMaxIdleTime"
+	connectionStringKey          = "connectionString"
+	ErrorMissingTableName        = "missing postgreSQL configuration table name"
+	InfoStartInit                = "Initializing PostgreSQL state store"
+	ErrorMissingConnectionString = "missing postgreSQL connection string"
+	ErrorAlreadyInitialized      = "PostgreSQL configuration store already initialized"
+	ErrorMissinMaxTimeout        = "missing PostgreSQL maxTimeout setting in configuration"
+	QueryTableExists             = "SELECT EXISTS (SELECT FROM pg_tables where tablename = $1)"
+)
+
+func NewPostgresConfigurationStore(logger logger.Logger) configuration.Store {
+	logger.Debug("Instantiating PostgreSQL configuration store")
+	return &ConfigurationStore{
+		logger: logger,
+	}
+}
+
+func (p *ConfigurationStore) Init(metadata configuration.Metadata) error {
+	p.logger.Debug(InfoStartInit)
+	if p.client != nil {
+		return fmt.Errorf(ErrorAlreadyInitialized)
+	}
+	if m, err := parseMetadata(metadata); err != nil {
+		p.logger.Error(err)
+		return err
+	} else {
+		p.metadata = m
+	}
+
+	ctx := context.Background()
```

Comment:
Please use a timeout with the context

Reply:
Added context with timeout

Reply:
https://github.com/dapr/components-contrib/blob/90ba8a441d6ccc8987acf115ca4adf1f49fa9fba/configuration/postgres/postgres.go#L102-L104

Reply:
Resolving this @ItalyPaleAle